File: src/test/java/com/github/kayjamlang/tests/containers/NamedExpressionFunctionContainerTest.java
package com.github.kayjamlang.tests.containers;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.containers.NamedExpressionFunctionContainer;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class NamedExpressionFunctionContainerTest {

    private static KayJamParser parser;

    @BeforeClass
    public static void prepare(){
        parser = new KayJamParser(new KayJamLexer("named fun test{}"));
    }

    @Test
    public void test() throws Exception {
        Expression expression = parser.readExpression();

        assertNotNull(expression);
        assertSame(NamedExpressionFunctionContainer.class, expression.getClass());

        NamedExpressionFunctionContainer namedExpressionFunction = (NamedExpressionFunctionContainer) expression;
        assertEquals(0, namedExpressionFunction.children.size());
    }
}

File: src/test/scala/com/github/kayjamlang/tests/containers/NamedExpressionFunctionContainerTest.scala
package com.github.kayjamlang.tests.containers

import com.github.kayjamlang.core.KayJamLexer
import com.github.kayjamlang.core.KayJamParser
import com.github.kayjamlang.core.containers.NamedExpressionFunctionContainer
import com.github.kayjamlang.core.expressions.Expression
import org.junit.BeforeClass
import org.junit.Test
import org.junit.Assert._


object NamedExpressionFunctionContainerTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("named fun test{}"))
  }
}

class NamedExpressionFunctionContainerTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = NamedExpressionFunctionContainerTest.parser.readExpression
    assertNotNull(expression)
    assertSame(classOf[NamedExpressionFunctionContainer], expression.getClass)
    val namedExpressionFunction = expression.asInstanceOf[NamedExpressionFunctionContainer]
    assertEquals(0, namedExpressionFunction.children.size)
  }
}
